src/libtiled/map.cpp

```diff
@@ -298,6 +298,18 @@ bool Map::addTemplateGroup(TemplateGroup *templateGroup)
     return true;
 }
 
+void Map::addTemplateGroupReference(const TemplateGroup *templateGroup)
+{
+//    mTemplateGroupReferences[templateGroup]++;
+    mTemplateGroupReferences.insert(templateGroup, mTemplateGroupReferences.value(templateGroup) + 1);
+}
+
+void Map::removeTemplateGroupReference(const TemplateGroup *templateGroup)
+{
+//    mTemplateGroupReferences[templateGroup]--;
```

Comment:
Did this not work for some reason?

Btw, while it makes sense to automatically remove template group references, this is only relevant when saving the map. So rather than keeping reference counts, the map writer could also just iterate over all the objects to determine the list of template group references that need to be written. I would prefer that approach.